packages/common/test/util.test.js

```diff
@@ -1,5 +1,365 @@
 import { expect } from 'chai';
-import { expandReferences, normalizeOauthConfig } from '../src/util';
+import {
+  enableMockClient,
+  expandReferences,
+  normalizeOauthConfig,
+  get,
+  post,
+  del,
+} from '../src/util';
+
+const client = enableMockClient('https://www.example.com');
+describe('request', () => {
+  it('should not set header content-type to application/json if body is string', async () => {
+    let request;
+    client
+      .intercept({
+        path: '/api',
+        method: 'POST',
+      })
+      .reply(200, r => {
+        request = r;
+        return '';
+      });
+
+    await post('/api', 'some string', {
+      baseUrl: 'https://www.example.com',
+    });
+
+    expect(request.headers).to.eql({});
+  });
+  it('should use baseUrl from options', async () => {
+    let request;
+    client
+      .intercept({
+        path: '/api',
+        method: 'GET',
+      })
+      .reply(200, r => {
+        request = r;
+        return {};
+      });
+
+    const response = await get('/api', { baseUrl: 'https://www.example.com' });
+
+    expect(request.path).to.eql('/api');
+    expect(response.code).to.eql(200);
+  });
+
+  it('should use fullUrlOrPath', async () => {
+    let request;
+    client
+      .intercept({
+        path: '/api',
+        method: 'GET',
+      })
+      .reply(200, r => {
+        request = r;
+        return {};
+      });
+
+    const response = await get('https://www.example.com/api');
+
+    expect(request.path).to.eql('/api');
+    expect(response.code).to.eql(200);
+  });
+});
+describe('methods', () => {
+  it('should send a GET', async () => {
+    // set up mock GET at /
+    let request;
+    client
+      .intercept({
+        path: '/api',
+        method: 'GET',
+      })
+      .reply(200, r => {
+        request = r;
+        return {};
+      });
+
+    await get('https://www.example.com/api');
+
+    // test the result
+    expect(request.method).to.eql('GET');
+  });
+
+  it('should send a POST', async () => {
+    // set up mock POST at /
+    let request;
+
+    client
+      .intercept({
+        path: '/api',
+        method: 'POST',
+      })
+      .reply(200, r => {
+        request = r;
```

Comment:
I know I drafted this pattern with you but I'm quickly going off it. It's a bit confusing in most cases. Mostly this pattern was created just suggest one way to test the `method` parameter of request.

I think we should set up an intercept a) with a unique URL and b) the things we want to test for in the intercept options.

If the request is correct and has the right headers, body, url, query etc, then the interceptor will match and we can just test the response code. Maybe the body too just to be sure that it's THIS interceptor that fired.

So this test would look more like this:
```
client
      .intercept({
        path: '/api/ali',
        method: 'POST',
      }).
     reply(200, 'ali-1')

const  { code, body } = await post('https://www.example.com/api/ali');

expect(code).to.eql(200)
expect(body).to.eql('ali-1');
```

Let's talk about this a bit more tomorrow.